src/item.cpp

```diff
@@ -11621,7 +11621,7 @@ int item::fill_with( const item &contained, const int amount,
 
     item contained_item( contained );
     const bool count_by_charges = contained.count_by_charges();
-    contained_item.charges = count_by_charges ? 1 : -1;
+    contained_item.charges = count_by_charges ? 1 : contained.charges;
```

Comment:
You should test that, but it doesn't seem to me like this line is necessary at all. Charges are set to 1 at the beginning of the loop anyway.
And if it's necessary, it should just be
```
if ( count_by_charges ) {
    contained_item.charges = 1;
}
```

Reply:
That is only for items that are counted by charges (ie. stacks of stuff like water).
Most items do not use charges (so it is usually 0). Some items use charges for other things (for example grenade ocuntdown).